enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,142 @@
+This will provide a more straightforward way to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Users will be able to remove any of the currently managed CVO objects by simply modifying the object's exisitng manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for managment by CVO therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO:
+
+```
+apiVersion: apps/v1
+...
+metadata:
+...
+  annotations:
+    release.openshift.io/delete: "true" 
+```
+
+### User Stories
+
+#### Story 1
+
+Remove the cluster-autoscaler-operator deployment.
+The existing cluster-autoscaler-operator deployment manifest 0000_50_cluster-autoscaler-operator_07_deployment.yaml is modified to contain the delete annotation:
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler-operator
+  namespace: openshift-machine-api
+  labels:
+    k8s-app: cluster-autoscaler-operator
+  annotations:
+    exclude.release.openshift.io/internal-openshift-hosted: "true"
+    release.openshift.io/delete: "true"
+...
+```
+
+#### Story 2
+
+In release 4.5 two jobs, openshift-service-catalog-controller-manager-remover and openshift-service-catalog-apiserver-remover, were created to remove the Service Catalog.
+For release 4.6 these jobs and all their supporting cluster objects should be removed.
+This enhancement allows a user to accomplish this in a very simple manner by modifying all the manifests associated with these two jobs to contain the delete annotation.
+In this case the manifests to be modified are:
+* 0000_50_cluster-svcat-apiserver-operator_04_roles.yaml
+* 0000_50_cluster-svcat-apiserver-operator_05_serviceaccount.yaml
+* 0000_50_cluster-svcat-controller-manager-operator_06_roles.yaml
+* 0000_50_cluster-svcat-controller-manager-operator_07_serviceaccount.yaml
+* 0000_90_cluster-svcat-apiserver-operator_01_remover_job.yaml
+* 0000_90_cluster-svcat-controller-manager-operator_01_remover_job.yaml
+
+### Implementation Details/Notes/Constraints
+
+CVO will use it's existing logic to discover and identify given manifests and to handle any associated errors.
+Once the manifest has been properly identified CVO will check for the delete annotation release.openshift.io/delete.
+If found and set to "true" the associated object will be removed from the cluster.
+If the annotation is set to "false" CVO will log a warning message.
+
+CVO will only support object removal for those objects that were originally created by CVO through its reconcilliation logic.
```

Comment:
nit: "reconcilliation" -> "reconciliation".

But I don't think we have a way to identify CVO-created resources and enforce this, so I'd rather drop the line.  Folks can easily audit specific release images to catch operators that misuse the power to remove some important, non-CVO manifest, and complain to the operator's maintainers about it.

Reply:
I guess it's sufficient to say that we're simply using the existing CVO logic to id a manifest and then look for the delete annotation.

Reply:
Changed in commit [262765d](https://github.com/openshift/enhancements/pull/419/commits/262765d0e846f11bb32d17fe4841e5b9917d4a82)